src/numeric-types.c

```diff
@@ -18,13 +18,15 @@
 
 #include "numeric-types.h"
 
+#include <glib.h>
 #include <string.h>
 
 #define DEFINE_NUMERIC(type)                                     \
-G_DEFINE_BOXED_TYPE (numeric_##type,                             \
-                     numeric_##type,                             \
-                     numeric_##type##_copy,                      \
-                     numeric_##type##_free)                      \
+G_DEFINE_BOXED_TYPE_WITH_CODE (numeric_##type,                   \
+                               numeric_##type,                   \
+                               numeric_##type##_copy,            \
+                               numeric_##type##_free,            \
+                               numeric_type_register_static (g_define_type_id, G_STRINGIFY(type), sizeof (numeric_##type), G_BYTE_ORDER)) \
```

Comment:
We have to add `byte_order` to the `DEFINE_NUMERIC` macro because not all definitions use host order.